client/src/components/SharingModal.vue

```diff
@@ -0,0 +1,241 @@
+                            <path
+                                d="M19 18.5C19 19.8807 17.8807 21 16.5 21C15.1193 21 14 19.8807 14 18.5C14 17.1193 15.1193 16 16.5 16C17.8807 16 19 17.1193 19 18.5Z"
+                                stroke="#1C274C" stroke-width="1.5" />
+                            <path
+                                d="M19 5.5C19 6.88071 17.8807 8 16.5 8C15.1193 8 14 6.88071 14 5.5C14 4.11929 15.1193 3 16.5 3C17.8807 3 19 4.11929 19 5.5Z"
+                                stroke="#1C274C" stroke-width="1.5" />
+                        </svg>
+
+                        <span>Share</span>
+                    </button>
+                    <button v-if="!value" class="button" @click="() => share(`Copy`)">
+                        <svg xmlns="http://www.w3.org/2000/svg" height="48" viewBox="0 0 24 24" width="48">
+                            <path d="M0 0h24v24H0V0z" fill="none" />
+                            <path
+                                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z" />
+                        </svg>
+                        <span>Copy</span>
+                    </button>
+                    <div v-if="value">
+                        <qrcode-vue :value="value" :level="level" :render-as="renderAs" />
+                    </div>
+                    <button :class="{ 'button':true, 'close-only': value  }"  @click="() => share(`close`)">
+                        <svg xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 24 24"
+                            fill="none">
+                            <path
+                                d="M20.7457 3.32851C20.3552 2.93798 19.722 2.93798 19.3315 3.32851L12.0371 10.6229L4.74275 3.32851C4.35223 2.93798 3.71906 2.93798 3.32854 3.32851C2.93801 3.71903 2.93801 4.3522 3.32854 4.74272L10.6229 12.0371L3.32856 19.3314C2.93803 19.722 2.93803 20.3551 3.32856 20.7457C3.71908 21.1362 4.35225 21.1362 4.74277 20.7457L12.0371 13.4513L19.3315 20.7457C19.722 21.1362 20.3552 21.1362 20.7457 20.7457C21.1362 20.3551 21.1362 19.722 20.7457 19.3315L13.4513 12.0371L20.7457 4.74272C21.1362 4.3522 21.1362 3.71903 20.7457 3.32851Z"
+                                fill="#0F0F0F" />
+                        </svg>
+                        <span>Close</span>
+                    </button>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { ref, defineProps, defineEmits } from 'vue';
+import QrcodeVue, { Level, RenderAs } from 'qrcode.vue';
+
+const value = ref('')
+const level = ref<Level>("H")
+const renderAs = ref<RenderAs>('svg')
+
+const props = defineProps({
+    title: {
+        type: String,
+        default: ''
+    },
+    subTitle: {
+        type: String,
+        default: ''
+    }
+});
+
+const emit = defineEmits(['share'])
+
+async function share(type: string) {
```

Comment:
I don't think the type is strictly needed here, it could just be different function calls for the different types.

Also I don't think we need to `emit('share', type);` this should probably just be emit('close'); or something similar.

Reply:
I removed this, was very odd looking sorry.